src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -472,52 +503,166 @@ private void CreateRegionLinks()
+        int priority = 0;
+        switch (shape)
+        {
+            case "lower":
+                intermediate1 = new Vector2(startCenter.X, lower.End.Y + i * 50);
+                intermediate2 = new Vector2(endCenter.X, lower.End.Y + i * 50);
+                priority = ReducePriority(start.Columns, start.Rows, priority);
+                priority = ReducePriority(end.Columns, start.Rows, priority);
+                break;
+            case "upper":
+                intermediate1 = new Vector2(startCenter.X, upper.Position.Y - i * 50);
+                intermediate2 = new Vector2(endCenter.X, upper.Position.Y - i * 50);
+                priority = ReducePriority(start.Columns, start.Rows, priority);
+                priority = ReducePriority(end.Columns, start.Rows, priority);
+                break;
+            case "right":
+                intermediate1 = new Vector2(right.End.X + i * 50, startCenter.Y);
+                intermediate2 = new Vector2(right.End.X + i * 50, endCenter.Y);
+                priority = ReducePriority(start.Rows, start.Columns, priority);
+                priority = ReducePriority(end.Rows, start.Columns, priority);
+                break;
+            case "left":
+                intermediate1 = new Vector2(left.Position.X - i * 50, startCenter.Y);
+                intermediate2 = new Vector2(left.Position.X - i * 50, endCenter.Y);
+                priority = ReducePriority(start.Rows, start.Columns, priority);
+                priority = ReducePriority(end.Rows, start.Columns, priority);
+                break;
+            default:
+                throw new Exception("Invalid side provided in GetUShapePathCharacteristics");
+        }
+
+        return (intermediate1, intermediate2, priority - i);
+    }
+
     /// <summary>
     ///   Get the least intersecting path from start region to end region. This is achieved by first calculating all
-    ///   possible paths, then figuring out which one intersects the least. If several paths are equally good, return
-    ///   the one with highest priority.
+    ///   possible paths, then figuring out which one has highest priority. If several paths are equally good, return
+    ///   the one with the least intersections.
     /// </summary>
     /// <remarks>
     ///   <para>
     ///     Priority: Direct path > L-shape path > Z-shape path > U-shape path
     ///   </para>
     /// </remarks>
     /// <returns>Path represented in a Vector2 array</returns>
-    private Vector2[] GetLeastIntersectingPath(PatchRegion start, PatchRegion end)
+    private (Vector2[] Path, bool StartVertical, bool EndVertical) GetLeastIntersectingPath(PatchRegion start,
+        PatchRegion end)
     {
         var startCenter = RegionCenter(start);
         var startRect = new Rect2(start.ScreenCoordinates, start.Size);
         var endCenter = RegionCenter(end);
         var endRect = new Rect2(end.ScreenCoordinates, end.Size);
+        int priority;
```

Comment:
Variables should be defined as close to where they are initialized as possible (so not all at the start of a method like ancient C styles mandated).